src/topology/config/mod.rs

```diff
@@ -14,7 +14,9 @@ mod vars;
 pub struct Config {
     #[serde(flatten)]
     pub global: GlobalOptions,
+    #[serde(default)]
```

Comment:
why do we need this?

Reply:
I want to be able to parse and validate configs that don't have sources or sinks (for our snippets work later on) so in order to do that I've defaulted the maps. The check then gets made explicit further down the path where we're building the topology: https://github.com/timberio/vector/pull/1064/files/e336e0bbb3ad187d70922a57f2e7afe195611926#diff-190025c16dded671e36ff22c0eedb95aR103

Reply:
Ah ok so defering to allowing serde to accept missing sources/sinks but then add a check further down 👍